package_nso_to_oc/xe/xe_system.py

```diff
@@ -1405,6 +1410,86 @@ def cleanup_name_server(config_leftover, name_server_list, vrf_list):
+        oc_system_inside_source.update(temp_nat_inside_source_list)
+
+def set_nat_pool(nat_pool, config_leftover):
+    pool_config = {"openconfig-system-ext:pool": []}
+    pool_config_list = pool_config["openconfig-system-ext:pool"]
+    # LIST OF NAT POOLS
+    for index, pool in enumerate(nat_pool):
+        start_address = end_address = "0.0.0.0" # Initialize variables
+        if pool.get("start-address"):
+            start_address = pool["start-address"]
+        if pool.get("end-address"):
+            end_address = pool["end-address"]
+        if pool.get("prefix-length"):
+            temp_pool = {"openconfig-system-ext:name": f'{pool.get("id")}',
+                        "openconfig-system-ext:config": {
+                            "openconfig-system-ext:name": f'{pool.get("id")}',
+                            "openconfig-system-ext:start-address": start_address,
+                            "openconfig-system-ext:end-address": end_address,
+                            "openconfig-system-ext:prefix-length": f'{pool.get("prefix-length")}'
+                        }}
+            pool_config_list.append(temp_pool)
+        elif pool.get("netmask"):
+            temp_pool = {"openconfig-system-ext:name": f'{pool.get("id")}',
+                        "openconfig-system-ext:config": {
+                            "openconfig-system-ext:name": f'{pool.get("id")}',
+                            "openconfig-system-ext:start-address": start_address,
+                            "openconfig-system-ext:end-address": end_address,
+                            "openconfig-system-ext:netmask": f'{pool.get("netmask")}'
+                        }}
+            pool_config_list.append(temp_pool)
+
+    del config_leftover["tailf-ned-cisco-ios:ip"]["nat"]["pool"]
+    return pool_config
+
+def set_nat_inside(nat_inside_source, config_leftover):
+    nat_inside = {"openconfig-system-ext:local-addresses-access-list": []}
+    nat_inside_list = nat_inside["openconfig-system-ext:local-addresses-access-list"]
+    # LIST OF NAT ACLs
+    if nat_inside_source.get("list"):
+        for inside_source in nat_inside_source.get("list"):
+            overload = False # Initialize variable
+            if "overload" in inside_source:
+                overload = True
+            temp_nat_inside = {"openconfig-system-ext:local-addresses-access-list-name": f'{inside_source.get("id")}', 
+                              "openconfig-system-ext:config": {
+                                "openconfig-system-ext:local-addresses-access-list-name": f'{inside_source.get("id")}',
+                                "openconfig-system-ext:global-interface-name": f'{inside_source.get("interface")}',
+                                "openconfig-system-ext:overload": overload
+                                }}
+            nat_inside_list.append(temp_nat_inside)
+    # LIST OF NAT ACLs WITH VRF
+    if nat_inside_source.get("list-vrf"):
```

Comment:
Can also skip this **if** with this:

```suggestion
    for inside_source in nat_inside_source.get("list-vrf", {}).get("list", []):
```